dns_server/src/crawler_p2p/crawler/address_data.rs

```diff
@@ -0,0 +1,326 @@
+                erase_after: _,
+                was_reachable,
+                fail_count: _,
+            } => *was_reachable,
+        }
+    }
+
+    /// Whether the address is currently recognized as reachable (available from DNS)
+    pub fn is_reachable(&self) -> bool {
+        match self {
+            AddressState::Connecting {
+                fail_count: _,
+                was_reachable: _,
+            } => false,
+            AddressState::Connected {} => true,
+            AddressState::Disconnecting {
+                fail_count: _,
+                was_reachable: _,
+            } => false,
+            AddressState::Disconnected {
+                fail_count: _,
+                was_reachable: _,
+                disconnected_at: _,
+            } => false,
+            AddressState::PermanentlyUnreachable {
+                fail_count: _,
+                was_reachable: _,
+                erase_after: _,
+            } => false,
+        }
+    }
+
+    /// Whether to retain the address between node restarts (store in DB)
+    pub fn is_persistent(&self) -> bool {
+        match self {
+            AddressState::Connecting {
+                fail_count: _,
+                was_reachable,
+            } => *was_reachable,
+            AddressState::Connected {} => true,
+            AddressState::Disconnecting {
+                fail_count: _,
+                was_reachable,
+            } => *was_reachable,
+            AddressState::Disconnected {
+                fail_count: _,
+                was_reachable,
+                disconnected_at: _,
+            } => *was_reachable,
+            AddressState::PermanentlyUnreachable {
+                fail_count: _,
+                was_reachable: _,
+                erase_after: _,
+            } => false,
+        }
+    }
+}
+
+impl AddressData {
+    /// Returns true when it is time to attempt a new outbound connection
+    pub fn connect_now(&self, now: Duration) -> bool {
+        match self.state {
+            AddressState::Connected {}
+            | AddressState::Connecting {
+                fail_count: _,
+                was_reachable: _,
+            }
+            | AddressState::Disconnecting {
+                fail_count: _,
+                was_reachable: _,
+            } => false,
+
+            AddressState::Disconnected {
+                fail_count,
+                was_reachable: _,
+                disconnected_at,
+            } if *self.user_added => {
+                // Try to connect to the user added nodes more often
+                let age = now - disconnected_at;
+                match fail_count {
+                    0 => true,
+                    1 => age > Duration::from_secs(60),
+                    2 => age > Duration::from_secs(360),
+                    _ => age > Duration::from_secs(3600),
+                }
+            }
+
+            AddressState::Disconnected {
+                fail_count,
+                was_reachable: true,
```

Comment:
Please create an if condition inside that the expression below. These splits can lead to issues in the future if we add more variables as tracking them is harder than just full arms.

Reply:
OK, updated.